### PR TITLE
New endpoint for getting last suggested document from chatkey 

### PR DESCRIPTION
**Pull request by daesolus**
Don't know why but my last commit was not included in the old pull request so I created a new one

### CODE REVIEW

File: WhisperAPI/WhisperAPI/Controllers/ContextController.cs [hunk 1]
```diff
@@ -21,20 +21,34 @@ public ContextController(IContexts contexts)
 
         public override void OnActionExecuting(ActionExecutingContext actionExecutingContext)
         {
-            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
-
-            log4net.ThreadContext.Properties["requestId"] = Guid.NewGuid();
             if (!this.ModelState.IsValid)
             {
                 actionExecutingContext.Result = this.BadRequest(this.ModelState);
-                Log.Error($"Search query:\r\n{JsonConvert.SerializeObject(searchQuery, Formatting.Indented)}");
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Controllers/ContextController.cs]: Is there a way we could keep this Log.Error ? Maybe we could do the TryGetValue first so we could have the searchQuery (if any).

File: WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs [hunk 2]
```diff
@@ -8,7 +9,8 @@ namespace WhisperAPI.Controllers
     [Route("/Whisper/[Controller]")]
     public class SuggestionsController : ContextController
     {
-        private static readonly log4net.ILog Log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+        private static readonly log4net.ILog Log =
+            log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs]: Why ?

File: WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs [hunk 3]
```diff
@@ -39,5 +41,18 @@ public IActionResult GetSuggestions([FromBody] SearchQuery searchQuery)
 
             return this.Ok(suggestion);
         }
+
+        [HttpGet]
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs]: HttpGet ? why not HttpPost like "public IActionResult GetSuggestions([FromBody] SearchQuery searchQuery)" ?

File: WhisperAPI/WhisperAPI.Tests/Integration/SuggestionsControllerTest.cs [hunk 4]
```diff
@@ -340,9 +373,42 @@ public ActionExecutingContext GetActionExecutingContext(SearchQuery searchQuery)
...
+                new List<IFilterMetadata>(),
+                new Dictionary<string, object>(),
+                this._suggestionController);
+
+            var param = new object();
+            actionExecutingContext
+                .Setup(x => x.ActionArguments.TryGetValue("searchQuery", out param))
+                .Returns(false);
+
+            param = chatkey;
```
[the remark this answers is not in the document]
Comment (daesolus) [reply WhisperAPI/WhisperAPI.Tests/Integration/SuggestionsControllerTest.cs]: TryGetValue need object type, chatkey is a guid so it cause an error.

File: WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs (hunk 2 above)
Comment (daesolus) [reply WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs]: ctrl+k+D

File: WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs (hunk 3 above)
Comment (daesolus) [reply WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs]: why post?, we query previous data, we do not save or update anything new

File: WhisperAPI/WhisperAPI/Controllers/ContextController.cs (hunk 1 above)
Comment (daesolus) [reply WhisperAPI/WhisperAPI/Controllers/ContextController.cs]: Why would we want to log bad user input data ? I understand when it is related to server data but model state is http 400, it is not a error on our part.

File: WhisperAPI/WhisperAPI.Tests/Integration/SuggestionsControllerTest.cs [hunk 5]
```diff
@@ -489,5 +467,20 @@ public List<string> GetIrrelevantIntents()
                 "Greetings"
             };
         }
+
+        private Mock<ActionExecutingContext> CreateActionContextMock()
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI.Tests/Integration/SuggestionsControllerTest.cs]: :+1:

**Pull request merged**